ntbs-service/DataAccess/NtbsContext.cs

```diff
@@ -252,29 +252,39 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
                 // it would completely clog up any EF actions. We've used manually created migrations instead.
             });
 
+            modelBuilder.Entity<PatientDetails>(entity =>
+            {
+                entity.HasKey(pd => pd.NotificationId);
+
+                entity.HasOne(pd => pd.PostcodeLookup)
+                    .WithMany()
+                    .HasForeignKey(ns => ns.PostcodeToLookup);
+
+                entity.ToTable("Patients");
+            });
+
+            modelBuilder.Entity<HospitalDetails>(entity =>
+            {
+                entity.HasKey(pd => pd.NotificationId);
+
+                entity.HasOne(hd => hd.CaseManager);
+
+                entity.ToTable("HospitalDetails");
+            });
+
             modelBuilder.Entity<Notification>(entity =>
             {
                 entity.HasOne(n => n.Group)
                     .WithMany(g => g.Notifications)
                     .HasForeignKey(e => e.GroupId);
 
-                entity.OwnsOne(e => e.HospitalDetails,
-                    hospitalDetails =>
-                    {
-                        hospitalDetails.HasOne(e => e.CaseManager);
-
-                        hospitalDetails.ToTable("HospitalDetails");
-                    });
+                entity.HasOne(n => n.PatientDetails)
+                    .WithOne()
+                    .OnDelete(DeleteBehavior.Cascade);
 
-                entity.OwnsOne(e => e.PatientDetails,
-                    x =>
-                    {
-                        x.HasOne(pd => pd.PostcodeLookup)
-                            .WithMany()
-                            .HasForeignKey(ns => ns.PostcodeToLookup);
-
-                        x.ToTable("Patients");
-                    });
+                entity.HasOne(n => n.HospitalDetails)
+                    .WithOne()
+                    .OnDelete(DeleteBehavior.Cascade);
 
                 entity.OwnsOne(e => e.ClinicalDetails,
```

Comment:
Is there a reason that clinical details is allowed to be owned? Is it because it contains no navigation properties?

Reply:
Yes. I was going for the smallest change necessary - there is still outstanding tech debt for removing the rest of the ownership relationships.